cli/internal/env/env.go

```diff
@@ -20,57 +68,54 @@ func getEnvMap() map[string]string {
 	return envMap
 }
 
-// getEnvPairsFromKeys returns a slice of key=value pairs for all env var keys specified in envKeys
-func getEnvPairsFromKeys(envKeys []string, allEnvVars map[string]string) []string {
-	hashableConfigEnvPairs := []string{}
-	for _, envVar := range envKeys {
-		hashableConfigEnvPairs = append(hashableConfigEnvPairs, fmt.Sprintf("%v=%v", envVar, allEnvVars[envVar]))
+// fromPrefixes returns a map of env vars and their values based on include/exclude prefixes
+func fromKeys(all EnvironmentVariableMap, keys []string) EnvironmentVariableMap {
+	output := EnvironmentVariableMap{}
+	for _, key := range keys {
+		output[key] = all[key]
 	}
 
-	return hashableConfigEnvPairs
+	return output
 }
 
-// getEnvPairsFromPrefix returns a slice of all key=value pairs that match the given prefix
-func getEnvPairsFromPrefix(includePrefix string, excludePrefix string, allEnvVars map[string]string) []string {
-	allEnvPairs := []string{}
-	for k, v := range allEnvVars {
-		if excludePrefix != "" && strings.HasPrefix(k, excludePrefix) {
-			continue
-		}
-		if strings.HasPrefix(k, includePrefix) {
-			allEnvPairs = append(allEnvPairs, fmt.Sprintf("%v=%v", k, v))
+// fromPrefixes returns a map of env vars and their values based on include/exclude prefixes
+func fromPrefixes(all EnvironmentVariableMap, includes []string, exclude string) EnvironmentVariableMap {
+	output := EnvironmentVariableMap{}
+	for _, prefix := range includes {
+		for k, v := range all {
+			// Skip vars that have the exclude prefix
+			if exclude != "" && strings.HasPrefix(k, exclude) {
+				continue
+			}
+
+			// if it has the prefix, include it
+			if strings.HasPrefix(k, prefix) {
+				output[k] = v
+			}
 		}
 	}
-	return allEnvPairs
+	return output
 }
 
-// getEnvPairsFromPrefixes returns a slice containing key=value pairs
-func getEnvPairsFromPrefixes(includePrefixes []string, excludePrefix string, allEnvVars map[string]string) []string {
-	allEnvPairs := []string{}
-	for _, includePrefix := range includePrefixes {
-		hashableFrameworkEnvPairs := getEnvPairsFromPrefix(includePrefix, excludePrefix, allEnvVars)
-		allEnvPairs = append(allEnvPairs, hashableFrameworkEnvPairs...)
-	}
-	return allEnvPairs
-}
+// Get returns all sorted key=value env var pairs for both frameworks and from envKeys
+func Get(keys []string, prefixes []string) DetailedMap {
```

Comment:
Not blocking: this is less descriptive than the old name in my opinion. `env.Get` could easily be confused with just getting all env vars.

Reply:
I was mostly getting rid of `GetHashableEnv "Pairs"`. `GetHashableEnvVars` seems better!